cameraview/src/main/java/com/otaliastudios/cameraview/Camera1.java

```diff
@@ -161,14 +161,43 @@ private boolean shouldStartPreview() {
         return isCameraAvailable() && mIsBound;
     }
 
+
+     void setDeviceOrientation(int deviceOrientation) {
+        super.setDeviceOrientation(deviceOrientation);
+        if (mCamera != null && mPreview != null) {
+            int cameraOrientation = getCameraOrientation();
+            mCamera.setDisplayOrientation(cameraOrientation);
+        }
+    }
+
+    public int getCameraOrientation(){
+        if (this.getDeviceOrientation() == 270)
+            return 0;
+        else if (this.getDeviceOrientation() == 90)
+           return 180;
+        else if (this.getDeviceOrientation() == 0)
+            return 90;
```

Comment:
Unfortunately, there is no guarantee for this to be the case. [The Nexus 5X has its camera sensor mounted the other way around](https://www.captechconsulting.com/blogs/android-camera-orientation-made-simple) (and who knows what's going on in other devices).

Reply:
Do you have any suggestion on how to improve this so that it gets approved?